api/application/services/protected_domain_service.py

```diff
@@ -90,16 +88,15 @@ def _verify_protected_domain_does_exist(self, domain):
             AppLogger.info(f"The protected domain, [{domain}] does not exist")
             raise UserError(f"The protected domain, [{domain}] does not exist.")
 
-    def _verify_protected_domain_is_empty(self, domain):
-        query = DatasetFilters(sensitivity=Sensitivity.PROTECTED)
-        datasets_metadata = self.resource_adapter.get_datasets_metadata(
-            s3_adapter=self.s3_adapter, query=query
-        )
-        datasets = [data.dataset for data in datasets_metadata if data.domain == domain]
+    def _verify_protected_domain_is_empty(
+        self, domain: str, schema_service: "SchemaService"
```

Comment:
Likewise with the comment above, we are now having a duplicate of the SchemaService